Model/Checkout/PlaceOrder.php

```diff
@@ -331,14 +331,19 @@ private function setCustomerEmail(CartInterface $quote, string $customerEmail):
         $this->quoteRepository->save($quote);
     }
 
-    private function getOrderByAmwalOrderId($amwalOrderId)
+    private function getOrderByAmwalOrderId($amwalOrderId, $quote)
     {
         // Build a search criteria to filter orders by custom attribute
         $searchCriteria = $this->searchCriteriaBuilder->addFilter('amwal_order_id', $amwalOrderId, 'eq');
         $searchCriteria = $searchCriteria->create();
 
         // Search for order with the provided custom attribute value and get the order data
         $order = $this->orderRepository->getList($searchCriteria)->getFirstItem();
+
+        if ($order->getEntityId()) {
+            $this->orderRepository->delete($order);
+            $order = $this->quoteManagement->submit($quote);
```

Comment:
I think this would cause duplicates, cause now we check if the order exists (has an ID), and if it does we submit the quote (creating a new order)

Reply:
Yes, we will create a new order for covering the calculation issues.